xla/service/gpu/runtime/nccl_clique.cc

```diff
@@ -180,6 +184,13 @@ static absl::Status CheckComm(NcclApi::NcclCommHandle comm) {
 // Runs async check on all communicators in a clique.
 static void CheckClique(const NcclCliqueKey& clique_key,
                         NcclClique& lockable_clique) {
+  if (TerminateOnNcclError()) {
+    absl::Status status = lockable_clique.CheckAsyncErrors();
+    if (!status.ok()) {
+      LOG(FATAL) << "Terminating process due to async NCCL error: " << status;
```

Comment:
Can we propagate Status instead?

Reply:
Do you mean propagating to the user (such as JAX)?

If yes, then we have such effort in the works, lead by @Tixxx . This PR is a workaround until proper propagation is in place. As things currently stand, you will have to kill the process anyway because ncclCommAbort can hang, so there is no reliable recovery.